telemetry_core/src/main/java/com/newrelic/telemetry/json/AttributesJson.java

```diff
@@ -1,36 +1,50 @@
 /*
  * ---------------------------------------------------------------------------------------------
- *  Copyright (c) 2019 New Relic Corporation. All rights reserved.
- *  Licensed under the Apache 2.0 License. See LICENSE in the project root directory for license information.
- * --------------------------------------------------------------------------------------------
+ *   Copyright (c) 2019 New Relic Corporation. All rights reserved.
+ *   Licensed under the Apache 2.0 License. See LICENSE in the project root directory for license information.
+ *  --------------------------------------------------------------------------------------------
  */
 package com.newrelic.telemetry.json;
 
+import java.io.IOException;
+import java.io.StringWriter;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.stream.Collectors;
 
-public interface AttributesJson {
+public class AttributesJson {
 
-  /**
-   * These attributes will be put in the common block, and this must be formatted like this to be
-   * accepted by the New Relic backend.
-   *
-   * <pre>
-   *   {
-   *     "stringKey": "StringValue",
-   *     "numberKey": 5555.88,
-   *     "booleanKey": true
-   *   }
-   * </pre>
-   */
-  String toJson(Map<String, Object> attributes);
+  public String toJson(Map<String, Object> attributes) {
+    StringWriter out = new StringWriter();
+    Map<String, Object> filteredAttributes = filterIllegalValues(attributes);
+    filterIllegalValues(filteredAttributes);
+    if (filteredAttributes.isEmpty()) {
+      return "{}";
+    }
+    try {
+      JsonWriter jsonWriter = new JsonWriter(out);
+      jsonWriter.beginObject();
 
-  /**
-   * Call this to remove illegal (NaN, -Infinity, null, etc) values from the attributes before
-   * generating JSON for them.
-   */
-  default Map<String, Object> filterIllegalValues(Map<String, Object> attributes) {
+      for (Map.Entry<String, Object> attribute : filteredAttributes.entrySet()) {
+        Object value = attribute.getValue();
+
+        if (value instanceof Boolean) {
+          jsonWriter.name(attribute.getKey()).value((boolean) value);
```

Comment:
the `jsonWriter.name(attribute.getKey())` bit is repeated in all cases.  We should pull it up.

Reply:
this wasn't changed from before. we can clean this up after this PR, I guess?